### ISSUE TITLE
PR 7370 relies on a commit outside of the visibility scope

### ISSUE THREAD

**Issue by lemenkov** [ledgerwatch/erigon#7383]
Hello! 
The same story as in the issue #5445. Looks like PR #5436 also relies on a ledgerwatch/erigon-lib@fc9dd74e6407 which is not yet merged into ledgerwatch/erigon-lib repository.

Note the banner "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository" on top of the page. Could you please address it? 

![Screenshot 2023-04-25 at 15-58-38 save · ledgerwatch_erigon-lib@fc9dd74](https://user-images.githubusercontent.com/41835/234300453-33818110-a234-40ed-9764-79ede6ac52fe.png)

It generates the following messages while compiling a 2.43.0 tag in a freshly cloned Erigon repo:

```
go: downloading github.com/mmcloughlin/addchain v0.4.0
go: downloading rsc.io/tmplfunc v0.0.3
../../core/blockchain.go:29:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../utils/flags.go:30:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../migrations/txs_begin_end.go:12:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../turbo/snapshotsync/block_snapshots.go:23:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../utils/flags.go:31:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../utils/flags.go:32:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
main.go:11:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/stagedsync/exec3.go:22:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../common/types.go:29:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../cl/cltypes/ssz/ssz.go:7:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/vm/gas_table.go:24:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../utils/flags.go:33:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../turbo/snapshotsync/block_snapshots.go:29:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/backend.go:45:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/backend.go:46:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../utils/flags.go:34:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../turbo/snapshotsync/block_snapshots.go:30:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/stagedsync/stage_snapshots.go:20:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/rw_v3.go:18:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../turbo/shards/state_change_accumulator.go:7:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/stagedsync/stage_snapshots.go:22:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../erigon-cl/execution_client/execution_client.go:11:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../ethdb/privateapi/all.go:7:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../turbo/shards/events.go:7:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../caplin-phase1/caplin1/run.go:6:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/protocols/eth/protocol.go:28:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../ethdb/privateapi/all.go:9:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/stagedsync/stage_finish.go:14:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../utils/flags.go:35:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/db_state_writer.go:12:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/temporal/kv_temporal.go:12:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/cached_reader2.go:9:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/plain_readonly.go:30:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/genesis_write.go:33:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../node/node.go:37:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/dump.go:27:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/genesis_write.go:34:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../rpcdaemon/cli/config.go:41:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../ethdb/privateapi/all.go:10:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/change_set_writer.go:10:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../turbo/snapshotsync/block_reader.go:14:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/types/block.go:34:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state/history_reader_inc.go:9:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state_transition.go:25:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../utils/flags.go:36:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../eth/backend.go:61:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407
../../core/state_transition.go:26:2: github.com/ledgerwatch/erigon-lib@v0.0.0-20230423044930-fc9dd74e6407: invalid version: unknown revision fc9dd74e6407

RPM build errors:
make: *** [Makefile:98: erigon.cmd] Error 1
```

**Comment by AskAlexSharov** [comment]
what is your golang version?
do you use go.work?

**Comment by lemenkov** [comment]
I use system-available Golang from my Linux distribution (which is 1.19.8 for Fedora 36/37 and 1.20.2 for Fedora 38/39 ) and do not rely on go.work. The thing is I just cannot pull this commit from the repo. GitHub stores all commits for the particular repo (including forks) inside one common internal database but if you clone the specific repo like https://github.com/ledgerwatch/erigon-lib/ you'll get only those commits which were pushed inside this one.

I am surprised nobody else stepped into this but right now you cannot fetch this ledgerwatch/erigon-lib@fc9dd74e6407 and it should be quite easy to reproduce:

```
Nostromo ~/work: git clone git@github.com:ledgerwatch/erigon-lib.git test-git
...
Nostromo ~/work: cd test-git/
Nostromo ~/work/test-git (git::main): git checkout -b test fc9dd74e6407
fatal: «fc9dd74e6407» не является коммитом, поэтому невозможно создать из него ветку «test»
Nostromo ~/work/test-git (git::main):
```

Golang just clones this repo using git and it cannot find the commit mentioned above.

**Comment by AskAlexSharov** [comment]
maybe old `git`?
can't reproduce on ubuntu-22.04 and we have CI on ubuntu-20.04 (go 1.19)
```
git version
git version 2.31.0

uname -a
Linux 5.19.0-1021-gcp #23~22.04.1-Ubuntu SMP Mon Apr 17 07:12:27 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

git clone git@github.com:ledgerwatch/erigon.git erigon_test
Cloning into 'erigon_test'...
cd erigon_test
make erigon
./build/bin/erigon --version
erigon version 2.44.0-dev-ded8283d


git checkout v2.43.0
make erigon
./build/bin/erigon --version
erigon version 2.43.0-stable-1106475e

rm ./build/bin/erigon
go clean -modcache
make erigon
./build/bin/erigon --version
erigon version 2.43.0-stable-1106475e


git config -l
core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
remote.origin.url=https://github.com/ledgerwatch/erigon.git
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
branch.devel.remote=origin
branch.devel.merge=refs/heads/devel
```

maybe related to go-mod-proxy? I see next config now:
```
go env | grep -i proxy
GONOPROXY=""
GOPROXY="https://proxy.golang.org,direct"
```

**Comment by lemenkov** [comment]
Fixed in PR #7432 so I am closing this. Erigon 2.44 [builds just fine](https://copr.fedorainfracloud.org/coprs/peter/Erigon/build/5999496/). Thanks!
